ts/features/barcode/types/__tests__/decoders.test.ts

```diff
@@ -166,36 +166,50 @@ describe("test decodeIOBarcode function", () => {
 
   describe("test SEND AAR barcode type", () => {
     const testUrls: Array<[string, boolean]> = [
-      ["https://dev.cittadini.notifichedigitali.it/?aar=whatever", true],
-      ["https://test.cittadini.notifichedigitali.it/?aar=whatever", true],
-      ["https://hotfix.cittadini.notifichedigitali.it/?aar=whatever", true],
-      ["https://uat.cittadini.notifichedigitali.it/?aar=whatever", true],
+      // "CITTADINI" domain
+      ["https://cittadini.uat.notifichedigitali.it/?aar=whatever", true],
       ["https://cittadini.notifichedigitali.it/?aar=whatever", true],
       [
         "https://cittadini.notifichedigitali.it/notifications/detail?aar=12345",
         true
       ],
       [
-        "  https://dev.cittadini.notifichedigitali.it/some/path?aar=abc123  ",
+        "  https://cittadini.uat.notifichedigitali.it/some/path?aar=abc123  ",
         true
       ],
-      ["https://dev.login.notifichedigitali.it/?aar=whatever", true],
-      ["https://test.login.notifichedigitali.it/?aar=whatever", true],
-      ["https://hotfix.login.notifichedigitali.it/?aar=whatever", true],
-      ["https://uat.login.notifichedigitali.it/?aar=whatever", true],
-      ["https://login.notifichedigitali.it/?aar=whatever", true],
+      ["https://cittadini.test.notifichedigitali.it/?aar=whatever", false],
+      ["https://cittadini.hotfix.notifichedigitali.it/?aar=whatever", false],
+      ["https://cittadini.dev.notifichedigitali.it/?aar=whatever", false],
+      // "LOGIN" domain
+      ["https://login.uat.notifichedigitali.it/?aar=whatever", true],
       [
         "https://login.notifichedigitali.it/notifications/detail?aar=12345",
         true
       ],
-      ["  https://dev.login.notifichedigitali.it/some/path?aar=abc123  ", true],
+      ["  https://login.uat.notifichedigitali.it/some/path?aar=abc123  ", true],
+      ["https://login.notifichedigitali.it/?aar=whatever", true],
+
+      ["https://login.test.notifichedigitali.it/?aar=whatever", false],
+      ["https://login.hotfix.notifichedigitali.it/?aar=whatever", false],
+      ["https://login.dev.notifichedigitali.it/?aar=whatever", false],
+      ["https://login.stage.notifichedigitali.it/?aar=whatever", false],
+      // invalid domains
+
       ["https://cittadini.notifichedigitali.it/?aar=", false],
       ["https://cittadini.notifichedigitali.it/?aar= ", false],
       ["https://login.notifichedigitali.it/?aar=", false],
       ["https://login.notifichedigitali.it/?aar= ", false],
       ["https://other-domain.it/?aar=whatever", false],
-      ["https://stage.cittadini.notifichedigitali.it/?aar=whatever", false],
-      ["https://stage.login.notifichedigitali.it/?aar=whatever", false]
+      ["https://cittadini.state.notifichedigitali.it/?aar=whatever", false],
```

Comment:
If we are checking for `cittadini.stage.notifiche...` then we should also check for `login.stage.notifiche...`

Reply:
it is checked! should be on line 196 👍